pystiche/papers/ulyanov_et_al_2016/data.py

```diff
@@ -0,0 +1,269 @@
+    "ulyanov_et_al_2016_style_transform",
+    "ulyanov_et_al_2016_images",
+    "ulyanov_et_al_2016_dataset",
+    "ulyanov_et_al_2016_batch_sampler",
+    "ulyanov_et_al_2016_image_loader",
+]
+
+
+def ulyanov_et_al_2016_content_transform(
+    edge_size: int = 256, impl_params: bool = True, instance_norm: bool = True,
+) -> ComposedTransform:
+    class OptionalGrayscaleToFakegrayscale(Transform):
+        def forward(self, input_image: torch.Tensor) -> torch.Tensor:
+            is_grayscale = extract_num_channels(input_image) == 1
+            if is_grayscale:
+                return grayscale_to_fakegrayscale(input_image)
+            else:
+                return input_image
+
+    transforms = []
+    if impl_params:
+        if instance_norm:
+            transforms.append(CenterCrop(edge_size))
+        else:
+            transforms.append(
+                Resize((edge_size, edge_size), interpolation_mode="bilinear")
+            )
+    else:
+        transforms.append(CenterCrop(edge_size))
+
+    transforms.append(OptionalGrayscaleToFakegrayscale())
+    return ComposedTransform(*transforms)
+
+
+def ulyanov_et_al_2016_style_transform(
+    impl_params: bool = True,
+    instance_norm: bool = True,
+    edge_size: Optional[int] = None,
+) -> ComposedTransform:
+    if edge_size is None:
+        edge_size = 256
+
+    if impl_params:
+        interpolation_mode = "bicubic" if instance_norm else "bilinear"
+    else:
+        interpolation_mode = "bilinear"
+
+    transforms = [
+        Resize(edge_size, edge="long", interpolation_mode=interpolation_mode)
+    ]
+    return ComposedTransform(*transforms)
+
+
+def ulyanov_et_al_2016_images(
+    root: Optional[str] = None, download: bool = True, overwrite: bool = False
+):
+
+    base_ulyanov = (
+        "https://raw.githubusercontent.com/DmitryUlyanov/texture_nets/master/data/"
+    )
+    base_ulyanov_suppl = "https://raw.githubusercontent.com/DmitryUlyanov/texture_nets/texture_nets_v1/supplementary/"
+    readme_ulyanov = "https://raw.githubusercontent.com/DmitryUlyanov/texture_nets/texture_nets_v1/data/readme_pics/"
+    content_base_ulyanov = urljoin(base_ulyanov, "readme_pics/")
+    content_images = {
+        "karya": DownloadableImage(
+            urljoin(content_base_ulyanov, "karya.jpg"),
+            md5="232b2f03a5d20c453a41a0e6320f27be",
+        ),
+        "tiger": DownloadableImage(
+            urljoin(content_base_ulyanov, "tiger.jpg"),
+            md5="e82bf374da425fb2c2e2a35a5a751989",
+        ),
+        "neckarfront": DownloadableImage(
+            "https://upload.wikimedia.org/wikipedia/commons/0/00/Tuebingen_Neckarfront.jpg",
+            title="Tübingen Neckarfront",
+            author="Andreas Praefcke",
+            license=CreativeCommonsLicense(("by",), version="3.0"),
+            md5="dc9ad203263f34352e18bc29b03e1066",
+            file="tuebingen_neckarfront__andreas_praefcke.jpg",
+        ),
+        "CheHigh": DownloadableImage(
```

Comment:
Please format all identifiers with lower case letters and separated with underscores.